common/src/main/java/org/wso2/testgrid/common/DashboardSetup.java

```diff
@@ -0,0 +1,142 @@
+    private String testplanID;
+
+    public DashboardSetup(String testplanID) {
+        this.testplanID = testplanID;
+    }
+
+    /**
+     * this method will create a DB in influxDB and add a new Data source to grafana
+     */
+    public void initDashboard() {
+        // create influxDB database according to tp_id
+        try {
+            InfluxDB influxDB = InfluxDBFactory.connect(restUrl + ":8086", username, password);
+            String dbName = testplanID;
+            influxDB.createDatabase(dbName);
+            influxDB.close();
+        } catch (AssertionError e) {
+            logger.error(StringUtil.concatStrings("Cannot create a new Database: ", e));
+        }
+        // add a new data source to grafana
+        HttpPost httpPost = createConnectivity(restUrl, apikey);
+        executeReq(getDataSource(testplanID), httpPost);
+
+    }
+    /**
+     * This method creates a json string that describes the grafana datasource
+     *
+     * @param name name of the data source that need to be created which is test plan ID
+     * @return string of jason string of the data source
+     */
+
+    private String getDataSource(String name) {
+
+        JSONObject user = new JSONObject();
+        user.put("name", name);
+        user.put("type", "influxdb");
+        user.put("url", "http://localhost:8086");
+        user.put("access", "proxy");
+        user.put("basicAuth", false);
+        user.put("password", "wso2123");
+        user.put("user", "telegraf");
+        user.put("database", name);
+
+        return user.toString();
+
+    }
+
+    /**
+     * this method will create the header of the POST request
+     * @param restUrl url of the grafana server
+     * @param apikey APIkey of the grafana server
+     * @return return a http post request
+     */
+
+    private HttpPost createConnectivity(String restUrl, String apikey) {
+        HttpPost post = new HttpPost(restUrl + ":3000/api/datasources/");
+        post.setHeader("AUTHORIZATION", apikey);
+        post.setHeader("Content-Type", "application/json");
+        post.setHeader("Accept", "application/json");
+        //post.setHeader("X-Stream" , "true");
+        return post;
+    }
+
+    /**
+     * this command will handle the posting http request to REST  api
+     * @param jsonData json string of the data source
+     * @param httpPost post request with authentication
+     */
+    private void executeReq(String jsonData, HttpPost httpPost) {
+        try {
+            executeHttpRequest(jsonData, httpPost);
+        } catch (UnsupportedEncodingException e) {
+            logger.error(StringUtil.concatStrings("error while encoding grafana http api url : ", e));
+        } catch (IOException e) {
+            logger.error(StringUtil.concatStrings("ioException occured while sending http request : ", e));
+        } catch (Exception e) {
+            logger.error(StringUtil.concatStrings("ioException occured while sending http request : ", e));
```

Comment:
Why do we need catch all Exceptions and log it as an ioException?

Reply:
Instead of saying ioException, lets add more context of the situation to the log message, so that a person reading the log can identify the exact situation the exception occurred.